pkg/storkctl/action.go

```diff
@@ -54,79 +56,29 @@ func newFailoverCommand(cmdFactory Factory, ioStreams genericclioptions.IOStream
 	var skipDeactivateSource bool
 	var includeNamespaceList []string
 	var excludeNamespaceList []string
-	var namespaceList []string
 	performFailoverCommand := &cobra.Command{
 		Use:   failoverCommand,
 		Short: "Initiate failover of the given migration schedule",
 		Run: func(c *cobra.Command, args []string) {
-			if len(referenceMigrationSchedule) == 0 {
-				util.CheckErr(fmt.Errorf("reference MigrationSchedule name needs to be provided for failover"))
-				return
-			}
-			// namespace of the migrationSchedule is provided by user using the -n / --namespace global flag
+			// namespace of the MigrationSchedule is provided by the user using the -n / --namespace global flag
 			migrationScheduleNs := cmdFactory.GetNamespace()
-			migrSchedObj, err := storkops.Instance().GetMigrationSchedule(referenceMigrationSchedule, migrationScheduleNs)
+			namespaceList, err := validationsForPerformDRCommands(storkv1.ActionTypeFailover, migrationScheduleNs, referenceMigrationSchedule, includeNamespaceList, excludeNamespaceList)
 			if err != nil {
-				util.CheckErr(fmt.Errorf("unable to find the reference MigrationSchedule %v in the %v namespace", referenceMigrationSchedule, migrationScheduleNs))
+				util.CheckErr(err)
 				return
 			}
-			if !skipDeactivateSource {
```

Comment:
One change I have made in the failover command is that clusterPair is required irrespective of skipDeactivateSource value. This is because even in skipDeactivateSource we would want to suspend the migrationSchedule in case source cluster is accessible

Reply:
But for unidirectional clusterpair and source not accessible, we still have to do the failover.

Reply:
But I cannot determine if source is accessible or not if clusterpair is not there ..
Maybe then the call to take is that if clusterpair not found in failover case we will assume source is inaccessible and will skip migrationSchedule suspension and source deactivation.
Or like we were discussing we can document that if unidirectional they should create reverse clusterpair with same name. And make having the cluster pair a pre-requisite

Reply:
As discussed offline we will be relaxing this check for failover.
If clusterpair is not provided we will just log a warning event and directly go ahead with scaleUpDestination. So in failover CLI we won't enforce anything wrt clusterpair

Reply:
We will need to discuss this in a meeting with rest of the team and finalise by end of this week. For now let's enforce it. I will make the change in another PR if required.